ipsource.go

```diff
@@ -337,20 +344,22 @@ func (Static) CaddyModule() caddy.ModuleInfo {
 	}
 }
 
-func (s Static) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP, error) {
-	if versions.V4Enabled() && versions.V6Enabled() {
+// Configured IP ranges are ignored, it doesn't make much sense to configure a
+// filter against a statically configured list of IPs.
+func (s Static) GetIPs(ctx context.Context, settings IPSettings) ([]net.IP, error) {
+	if settings.V4Enabled() && settings.V6Enabled() {
 		return s.IPs, nil
 	}
 
 	ips := []net.IP{}
 
 	for _, ip := range s.IPs {
-		if versions.V4Enabled() && ip.To4() != nil {
+		if settings.V4Enabled() && ip.To4() != nil {
 			ips = append(ips, ip)
 			continue
 		}
 
-		if versions.V6Enabled() && ip.To16() != nil {
+		if settings.V6Enabled() && ip.To4() == nil {
```

Comment:
I left a long comment in the commit message for why I did this, but now I realise it wasn't necessary, since V4 IPs will hit the continue in the previous if block and not reach here.